app/domain/authentication/authn_k8s/k8s_object_lookup.rb

```diff
@@ -134,12 +134,29 @@ def invoke_k8s_method method_name, *arguments
         k8s_client_for_method(method_name).send(method_name, *arguments)
       end
 
+      def k8s_client_info client
+        group    = client.instance_variable_get(:@api_group)
+        version  = client.instance_variable_get(:@api_version)
+        entities = client.instance_variable_get(:@entities).to_yaml
+        "api_group: #{group}, api_version: #{version}, entities: #{entities}"
+      end
+
       # Methods move around between API versions across releases, so search the
       # client API objects to find the method we are looking for.
       def k8s_client_for_method method_name
         k8s_clients.find do |client|
           begin
-            client.respond_to?(method_name)
+            responded   = client.respond_to?(method_name)
+            # The call to respond_to? has updated the list of entities
+            # supported by that client, so we can gather that now.
+            client_info = k8s_client_info(client)
+            result = if responded
+              "supported"
+            else
+              "NOT supported"
+            end
```

Comment:
ruby =/= Go :)

A more idiomatic way to write this would be:
```ruby
support_desc = responded ? "supported" : "NOT supported"
```

Reply:
This is what happens when a Go coder tries writing Ruby.